source/ui/CLS/CLSBeamlineStatusView.h

```diff
@@ -54,26 +62,43 @@ protected slots:
 	QWidget* createFullBeamlineStatusView();
 	/// creates and returns a layout with the component buttons
 	QLayout* createBeamlineStatusButtonBarLayout();
+
+	/// creates and returns a layout with the beam on/off buttons
+	QLayout* createBeamOnOffButtons();
+
 	/// Creates and returns a component view for the given control.
 	virtual QWidget* createComponentView(AMControl *control);
 
+	/// helper functions to create a push button
+	QPushButton *createPushButton(const QString &text);
+
 protected:
 	/// flag to identify whether this is the compact status view or full status view
 	bool compactView_;
+
+	/// The beamline being operated
+	CLSBeamline *beamline_;
```

Comment:
This is unnecessary because there will only ever be one beamline which can always be accessed by `CLSBeamline::bl()`.